src/api/firebase.js

```diff
@@ -246,3 +246,45 @@ export async function deleteItem() {
 	 * this function must accept!
 	 */
 }
+
+export function comparePurchaseUrgency(a, b) {
+	const today = new Date();
+	const nextPurchaseDaysBetween = Number(
+		getDaysBetweenDates(today, a.dateNextPurchased.toDate()),
+	);
+	const lastPurchaseDaysBetween = getDaysBetweenDates(
+		a.dateLastPurchased[a.dateLastPurchased.length - 1].toDate(),
+		today,
+	);
+	const nextPurchaseDaysBetweenB = Number(
+		getDaysBetweenDates(today, b.dateNextPurchased.toDate()),
+	);
+	const lastPurchaseDaysBetweenB = getDaysBetweenDates(
+		b.dateLastPurchased[b.dateLastPurchased.length - 1].toDate(),
+		today,
+	);
+	// if not purchased within 60 days, sort to bottom of list
+	if (lastPurchaseDaysBetween > 60 && lastPurchaseDaysBetweenB < 60) {
+		return 1;
+	}
+	if (lastPurchaseDaysBetween < 60 && lastPurchaseDaysBetweenB > 60) {
+		return -1;
+	}
+	// sort by days until next purchase
+	if (nextPurchaseDaysBetween < nextPurchaseDaysBetweenB) {
+		return -1;
+	}
+	// if days until next purchase is the same, sort alphabetically
+	if (nextPurchaseDaysBetween === nextPurchaseDaysBetweenB) {
+		if (a.name < b.name) {
+			return -1;
+		} else if (a.name === b.name) {
+			return 0;
+		} else {
+			return 1;
+		}
+	}
+	if (nextPurchaseDaysBetween > nextPurchaseDaysBetweenB) {
+		return 1;
+	}
+}
```

Comment:
praise: great job for implementing this. It's not easy!

refactor: I want to make it simpler, but it's not required to refactor it. Make sure to understand the code first before refactoring it. `.sort()` is quite confusing if we don't understand the basic.

```js
export function comparePurchaseUrgency(a, b) {
	const today = new Date();
	const nextPurchaseDaysBetween = Number(
		getDaysBetweenDates(today, a.dateNextPurchased.toDate()),
	);
	const lastPurchaseDaysBetween = getDaysBetweenDates(
		a.dateLastPurchased[a.dateLastPurchased.length - 1].toDate(),
		today,
	);
	const nextPurchaseDaysBetweenB = Number(
		getDaysBetweenDates(today, b.dateNextPurchased.toDate()),
	);
	const lastPurchaseDaysBetweenB = getDaysBetweenDates(
		b.dateLastPurchased[b.dateLastPurchased.length - 1].toDate(),
		today,
	);
	// if not purchased within 60 days, sort to bottom of list
	if (lastPurchaseDaysBetween > 60 && lastPurchaseDaysBetweenB < 60) {
		return 1;
	} else if (lastPurchaseDaysBetween < 60 && lastPurchaseDaysBetweenB > 60) {
		return -1;
	}

	// if days until next purchase is the same, sort alphabetically
	if (
		nextPurchaseDaysBetween === nextPurchaseDaysBetweenB ||
		(lastPurchaseDaysBetween > 60 && lastPurchaseDaysBetweenB > 60)
	) {
		return a.name.localeCompare(b.name);
	}

	// sort by days until next purchase
	return nextPurchaseDaysBetween - nextPurchaseDaysBetweenB;
}
```

I also added another condition to compare the name alphabetically `(lastPurchaseDaysBetween > 60 && lastPurchaseDaysBetweenB > 60)`. I think we missed to handle the alphabetical order when we more than one inactive item

Reply:
I added a condition to sort the inactive items by name. I didn't implement your refactoring suggestion yet though, as I didn't have time to understand it fully and add it to our code. I made a new issue for refactoring code in our firebase file, and added it on so that we can revisit.